packages/native/src/useLinking.tsx

```diff
@@ -40,6 +40,21 @@ export default function useLinking(
   ref: React.RefObject<NavigationContainerRef>,
   { prefixes, config, getStateFromPath = getStateFromPathDefault }: Options
 ) {
+  React.useEffect(() => {
+    const listenerRef = ref?.current?.__isListenerSetRef;
```

Comment:
We need to use a variable in file scope to do it. Such a ref will handle 2 `useLinking` used for the same container, but if there were 2 containers (could be in a brownfield app), then we wouldn't catch it.